model/DataDifferencer.php

```diff
@@ -197,26 +199,29 @@ public function ChangedFields() {
 			if(!$this->fromRecord || $this->fromRecord->$field != $this->toRecord->$field) {
 				// Only show HTML diffs for fields which allow HTML values in the first place
 				$fieldObj = $this->toRecord->dbObject($field);
-				if($this->fromRecord) {
-					$fieldDiff = Diff::compareHTML(
-						$this->fromRecord->$field,
-						$this->toRecord->$field,
-						(!$fieldObj || $fieldObj->stat('escape_type') != 'xml')
-					);
-				} else {
-					if($fieldObj && $fieldObj->stat('escape_type') == 'xml') {
-						$fieldDiff = "<ins>" . $this->toRecord->$field . "</ins>";
+
+				if($fieldObj instanceof DBField) {
```

Comment:
`$this->record` on DataObject is populated by `DataObject::setField()`, which will be called by any setter, including injected `$depedencies`.

I think that the fix here is to return null if `dbObject()` is called to access something that isn't a DBField. It may even be beneficial to make it stricter, and only return properties that have been sourced from the database.